### PR TITLE
[chore] instruction for installing mava for developement

### PR DESCRIPTION
**Pull request by sash-a**
## What?
Added install instruction for people who want to develop for mava to readme.
## Why? 
I couldn't find this information easily and think it should be in the install instructions
## How?
Changed the readme

### CODE REVIEW

**Review by ulricharmel — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -72,6 +72,13 @@ You can also install directly from source:
 pip install "id-mava[reverb,jax,envs] @ git+https://github.com/instadeepai/mava.git"
 ```
 
+If you would like to develop features for Mava, the best way to do download all dependencies is:
+
+```bash
+git clone https://github.com/instadeepai/mava.git
+pip install -e "mava[reverb,jax,envs]"
```
Comment (RuanJohn) [inline README.md:79]: Maybe don't have all the envs here. Just have `[reverb, jax]` and then add an environment specific example to the detailed install section please?

File: DETAILED_INSTALL.md [hunk 2]
```diff
@@ -203,6 +203,17 @@ docker run --gpus all -it --rm  -v $(pwd):/home/app/mava -w /home/app/mava insta
...
+ 
+        When developing mava locally one can just replace the `id-mava` part of the `pip install` with a `.`.  
+        For example when installing petting zoo dependencies for local development (inside of the mava folder):
+
+        ```bash
+        pip install -e ".[pz]"
+        ```
+
+        This installs the petting zoo mava dependencies into your current environment, but uses your local copy of mava, instead of one stored inside of your environments package directory.
+
```
Comment (OmaymaMahjoub) [inline DETAILED_INSTALL.md:216]: Probably we need to keep it outside the bullet list of env by making it as a note:

> - Development
  When developing mava locally one can just replace the `id-mava` part of the `pip install` with a `.`.  
   For example when installing petting zoo dependencies for local development (inside of the mava folder):
>   ```bash
>   pip install -e ".[pz]
>   ```
>   This installs the petting zoo mava dependencies into your current environment, but uses your local copy of mava, instead of one stored inside of your environments package directory.

**Review by OmaymaMahjoub — DISMISSED** [review]
Thank you @sash-a, this will be very helpful instructions, just an optional request that it will make the docs more suitable!

**Review by RuanJohn — APPROVED** [review]
Thanks @sash-a :muscle:

File: README.md [hunk 3]
```diff
@@ -72,6 +72,13 @@ You can also install directly from source:
 pip install "id-mava[reverb,jax,envs] @ git+https://github.com/instadeepai/mava.git"
 ```
 
+If you would like to develop features for Mava, the best way to do download all dependencies is:
```
Comment (RuanJohn) [inline README.md:75]: minor fix. 
"If you would like to develop features for Mava, please install locally from source as follows:"

File: DETAILED_INSTALL.md [hunk 4]
```diff
@@ -203,6 +203,17 @@ docker run --gpus all -it --rm  -v $(pwd):/home/app/mava -w /home/app/mava insta
 
         If this fails, follow instructions [here](https://github.com/deepmind/meltingpot#installation).
 
+    * Development
+ 
+        When developing mava locally one can just replace the `id-mava` part of the `pip install` with a `.`.  
+        For example when installing petting zoo dependencies for local development (inside of the mava folder):
```
Comment (KaleabTessera) [inline DETAILED_INSTALL.md:209]: I think we can simplify this part, the "inside of the mava folder" might not be clear for new users. 

Maybe rephrase to something like,
For example, if you would to install a local version of mava, including the petting zoo dependencies...

I also think the content added to the readme, should be placed here. We should keep the readme simple and here we can we can keep all the instructions to development on mava. Maybe we can also link to this page in the Contributing docs.

**Review by KaleabTessera — CHANGES_REQUESTED** [review]

File: DETAILED_INSTALL.md (hunk 4 above)
Comment (sash-a) [reply DETAILED_INSTALL.md:209]: Agreed, I'll remove it from the readme and I'll added a section in the contribution docs with a similar example

**Review by RuanJohn — DISMISSED** [review]
Thanks @sash-a :book:

File: DETAILED_INSTALL.md [hunk 5]
```diff
@@ -181,7 +181,34 @@ docker run --gpus all -it --rm  -v $(pwd):/home/app/mava -w /home/app/mava insta
...
+        ```
+
+        Add MeltingPot to your python path:
+
+        ```bash
+        export PYTHONPATH="${PYTHONPATH}:${PWD}/../packages/meltingpot"
+        ```
+
+        If this fails, follow instructions [here](https://github.com/deepmind/meltingpot#installation).
+
```
Comment (OmaymaMahjoub) [inline DETAILED_INSTALL.md:199]: I think there was merging conflict if you can delete this section.

File: DETAILED_INSTALL.md (hunk 2 above)
Comment (sash-a) [reply DETAILED_INSTALL.md]: Done :smile:

File: DETAILED_INSTALL.md (hunk 5 above)
Comment (sash-a) [reply DETAILED_INSTALL.md]: Done :smile:

File: DETAILED_INSTALL.md [hunk 6]
```diff
@@ -181,7 +181,20 @@ docker run --gpus all -it --rm  -v $(pwd):/home/app/mava -w /home/app/mava insta
         pip install id-mava[sc2]
         ```
 
-3. Run an example:
+        If this fails, follow instructions [here](https://github.com/deepmind/meltingpot#installation).
```
Comment (OmaymaMahjoub) [inline DETAILED_INSTALL.md:184]: If you can delete this too @sash-a!

Comment (sash-a) [reply DETAILED_INSTALL.md:184]: my bad :facepalm:

**Review by OmaymaMahjoub — APPROVED** [review]
Thanks @sash-a :raised_hands:

**Review by OmaymaMahjoub — COMMENTED** [review]
Why python 3.9 check failed?

**Review by KaleabTessera — APPROVED** [review]
Thanks @sash-a !
